library/swift/test/RetryPolicyMapperTests.swift

```diff
@@ -23,4 +23,9 @@ final class RetryPolicyMapperTests: XCTestCase {
     let policy = RetryPolicy(maxRetryCount: 123, retryOn: RetryRule.allCases)
     XCTAssertNil(policy.outboundHeaders()["x-envoy-upstream-rq-per-try-timeout-ms"])
   }
+
+  func testConvertingToHeadersWithoutUpstreamTimeoutIncludesZeroForTimeoutHeader() {
```

Comment:
let's add a test case for when per request timeout is equal to 0 (not nil) and per retry timeout is greater than 0.

Reply:
Considered that but iOS doesn't really provide a way to test that `assert`s fail, and making the builder `throw` doesn't really seem worth it here